### PR TITLE
New distributive packages

### LINKED ISSUE #1259: distributive symlink failures
Seeing the following errors when running a build:

```
TASK: [consul | create symlink to distributive consul checklist] **************
failed: [resching-aws-worker-03] => {"failed": true, "path": "/etc/distributive.d/consul.json", "src": "/usr/share/distributive/consul.json", "state": "absent"}
msg: src file does not exist, use "force=yes" if you really want to create the link: /usr/share/distributive/consul.json
...
```

It looks like this was discussed in gitter but documenting it here for anyone else who runs into the issue.

If a build fails, you can workaround it for now by running:

```
ansible all -s -m yum -a 'name=distributive state=absent'
ansible all -s -m yum -a 'name=distributive-0.2.4-1.x86_64 state=present'
```

and then restarting your build.

- Ansible version (`ansible --version`): 1.9.4 
- Python version (`python --version`): 2.7.6
- Git commit hash or branch: master
- Cloud Environment: aws
- Terraform version (`terraform version`): v0.6.11

### PR DESCRIPTION
**Pull request by siddharthist**
- [x] Installs cleanly on a fresh build of most recent master branch
- [ ] Upgrades cleanly from the most recent release
- [x] Updates documentation relevant to the changes

Upgrade to new packaging system, which will make fixing broken checks a vastly faster process. 
Removes duplicated code from back when I didn't know how to Ansible. 
Fixes #1259

The dnsmasq checks are failing on this because of #1265

### CODE REVIEW

**Comment by siddharthist** [comment]
This will be ready as soon as https://github.com/asteris-llc/mantl-packaging/pull/56 is merged. This has been tested on DO, Vagrant.

**Comment by ryane** [comment]
it looks like https://github.com/asteris-llc/mantl-packaging/pull/56 has been merged but I'm still seeing  broken distributive checks (upgrade from 1.0.3 on aws):

```
time="2016-03-15T14:21:28Z" level=info msg="Report from checklist" checklist="dnsmasq health check" report="↴\nTotal: 8\nPassed: 7\nFailed: 1\nOther: 0\nIs not a file: /etc/resolv.conf" 
```

**Comment by siddharthist** [comment]
@ryane The CI build is broken on mantl-packaging, and so didn't publish the new version of that RPM. I'll do so manually soon.

**Comment by siddharthist** [comment]
Actually, [it looks like it has been uploaded](https://bintray.com/asteris/mantl-rpm/distributive-dnsmasq/view). I'll do some digging soon.

**Comment by ryane** [comment]
@siddharthist I just revisited this. I noticed a couple of things. First, the distributive package itself is not being upgraded. After applying this pr on 1.0.3, I end up with:

`distributive.x86_64                  0.2.4-1`

On a fresh build, I end up with:

`distributive.x86_64                0.2.5-1`

Also, all of the `distributive*` files in `/etc/consul` are still present post-upgrade so that also generates a bunch of failed checks.

**Comment by siddharthist** [comment]
@ryane Thanks for checking, I think that change should fix the issue with the distributive package being upgraded. As for deregistering checks, do you have any suggestions? We could just make an upgrade ansible playbook, but that's kinda messy.

File: roles/distributive/tasks/main.yml [hunk 1]
```diff
@@ -15,7 +15,7 @@
 - name: install distributive, health check packages for current role
   become: yes
   yum:
-    state: present
+    state: latest
```
Comment (ryane) [inline roles/distributive/tasks/main.yml]: if we do `latest`, though, aren't we opening ourselves up for problems down the road if we release a new version of distributive that is incompatible with this release? this is what happened with mantl-dns in #1265. Can we just explicitly tell it to install `0.2.5-1`?

**Comment by ryane** [comment]
yeah, it is messy. we do already have a precedent of upgrade logic in playbooks: https://github.com/CiscoCloud/mantl/blob/master/roles/zookeeper/tasks/main.yml#L33. It's not ideal but not sure how to get around it in this case.

File: roles/distributive/tasks/main.yml (hunk 1 above)
Comment (siddharthist) [inline roles/distributive/tasks/main.yml]: We were having that discussion in Slack the other day. That's my preferred option, I'll make it so.

**Comment by siddharthist** [comment]
All the versions are now pinned to avoid pain later. I'll work on an upgrade playbook.

**Comment by siddharthist** [comment]
Looks like `dnsmasq-0.2.5-1` contains the checklist from `dnsmasq-0.2.4-1`! I have no idea how this happened, but I'm going to delete the erronious package and upload the correct one.

File: roles/distributive/tasks/main.yml [hunk 2]
```diff
@@ -8,10 +8,34 @@
     dest: /etc/sudoers.d/consul-docker-auth
     validate: visudo -c -f "%s"
   tags:
+    - bootstrap
     - consul
     - distributive
     - docker
 
+# Needed for upgrade from Mantl <1.0
```
Comment (ryane) [inline roles/distributive/tasks/main.yml]: super minor but it is actually needed for upgrading mantl 1.0.x to 1.1

File: roles/distributive/tasks/main.yml [hunk 3]
```diff
@@ -8,10 +8,34 @@
...
+  file:
+    path: "/etc/consul/distributive-{{ current_role }}-check.json"
+    state: absent
+  notify:
+    - reload consul
+  tags:
+    - bootstrap
+    - consul
+    - distributive
+
```
Comment (ryane) [inline roles/distributive/tasks/main.yml]: I tried upgrading a 1.0.3 cluster with this and I still end up with the distributive checks in the consul directory and the distributive directory:

```
$ sudo ls -al /etc/consul
total 72
drwxrwx---.  3 consul consul 4096 Mar 17 20:32 .
drwxr-xr-x. 95 root   root   8192 Mar 17 20:22 ..
-rw-r--r--.  1 consul consul  163 Mar 17 19:57 acl.json
-rw-r--r--.  1 root   root     57 Mar 17 19:58 agent_acl.json
-rw-rw-r--.  1 consul consul  753 Mar 17 20:21 consul.json
-rw-r--r--.  1 root   root    995 Mar 17 19:59 consul.nginx
-rw-r--r--.  1 root   root     41 Jan 19 14:51 consul-ui.json
-rw-r--r--.  1 consul consul  276 Mar 17 19:52 distributive-common-check.json
-rw-r--r--.  1 root   root    296 Mar 17 20:32 distributive-consul-check.json
-rw-r--r--.  1 root   root    299 Mar 17 20:32 distributive-dnsmasq-check.json
-rw-r--r--.  1 root   root    302 Mar 17 20:32 distributive-logstash-check.json
-rw-r--r--.  1 root   root    299 Mar 17 20:32 distributive-mantlui-check.json
-rw-r--r--.  1 root   root    333 Mar 17 20:32 distributive-zookeeper-check.json
-rw-r--r--.  1 consul consul  176 Mar 17 19:57 docker-service.json
-rw-r--r--.  1 root   root    312 Mar 17 20:02 marathon.json
drwxrwxr-x.  2 consul consul   55 Mar 17 19:57 ssl
-rw-r--r--.  1 root   root    261 Mar 17 20:00 zk-consul.json
```

```
ls -al /etc/distributive.d/
total 36
drwxr-xr-x.  2 root root 4096 Mar 17 20:24 .
drwxr-xr-x. 95 root root 8192 Mar 17 20:22 ..
lrwxrwxrwx.  1 root root   35 Mar 17 19:52 common.json -> /usr/share/distributive/common.json
-rw-rw-r--.  1 root root 1566 Feb  9 21:54 distributive-consul.json
-rw-rw-r--.  1 root root 1336 Mar 14 22:01 distributive-dnsmasq.json
-rw-rw-r--.  1 root root 1178 Mar 11 14:04 distributive-logstash.json
-rw-rw-r--.  1 root root  910 Mar 11 19:28 distributive-mantlui.json
-rw-rw-r--.  1 root root 1213 Feb  9 21:57 distributive-zookeeper.json
```

There were no errors during the ansible run. And all nodes are unhealthy b/c of this after the upgrade. Any idea?

**Comment by ryane** [comment]
@siddharthist do you think we should merge this as-is and then create a new issue to track the upgrade to 1.1? It would be good to get master fixed

**Comment by ryane** [comment]
this fixes new builds but upgrades are still a problem. tracking in #1288

**Pull request merged**